stackrox/stackrox
# ROX-19530: Add conditional rendering in AuthenticationTokensSection

## Description

### Problems

1. Integrations page renders Cluster Init Bundles even if not enabled according to user role permissions.
2. Integration sagas cause 403 Forbidden status if the route is not enabled in general and if Cluster Init Bundles is not enabled in specific.

### Analysis

1. Cluster Init Bundles tile
    * differs in resource access requirements from other integration tiles.
    * will differ in link after the table and form move from integrations to clusters route.
    * will not cause 403 Forbidden status if it encapsulates its request for integration count.
    * will be deprecated for future removal.
2. Unlike `tableColumnDescriptors` which are accessed according to source and type from page address, it is possible and simpler for TypeScript to separate `integrationsList` into separate arrays for each source. The only dynamic access is for label.

### Solution

1. Refactor IntegrationTilesPage.tsx file:
    * Delegate responsibility for data to source sections, similar to other dashboards. Future update will replace sagas with thunk to make requests.
    * Move feature flag filter and counts function to integrationTiles.utils.ts so source sections can import it.
2. Export descriptor arrays by source from integrationsList.ts file.
3. Move `GalleryItem` element to `IntegrationTile` element so `IntegrationsSection` renders its children directly.
4. Bonus markup
    * Add `component="div"` prop to `PageSection` for `h1` element, because it is for presentation instead of page structure.
    * Ditto for sibling `PageSection` element for integration source sections.

## Checklist

## Testing Performed

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: -4 = 3763528 - 3763532
        total: 146 = 10055512 - 10055366
    * `ls -al build/static/js/*.js | wc -l`
        0 = 81 - 81 files
3. `yarn start` in ui

### Manual testing

1. Visit /main/integrations with **Admin** role
    * See extra GET /v1/cluster-init/init-bundles request from `ClusterInitBundlesTile` element in addition to sagas, which we will remove soon.
    * Click tiles and links in tables to make sure refactoring did not break anything.

2. Visit /main/integrations with role that has **Analyst** permission set, therefore NO_ACCESS for Administration resource
    * See 403 Forbidden status for GET /v1/cluster-init/init-bundles from sagas.
    * See absence of extra request because conditional rendering omits `ClusterInitBundlesTile` element.

![AuthenticationTokensSection](https://github.com/stackrox/stackrox/assets/11862657/a6c915be-dced-468a-8fba-dc6a48a5d1b0)

### Integration testing

* integrations/apiTokens.test.js
* integrations/clusterInitBundles.test.js
* integrations/externalBackups.test.js
* integrations/general.test.js
* integrations/imageIntegrations.test.js
* integrations/notifiers.test.js
* integrations/signatureIntegrations.test.js
